### PR TITLE
Add appmap models

### PR DESCRIPTION
**Pull request by dustinbyrne**
Add Appmap models, complete with client-side normalization and event pruning.

```js
const appmap = buildAppMap()
      .source(await fromFile(file))
      .prune(cmd.size)
      .normalize()
      .build();
```

### CODE REVIEW

**Review by apotterri — APPROVED** [review]
LGTM. Probably needs a build job set up soon.

File: src/appmapBuilder.js [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+      return chunks;
+    }, []);
+  }
+}
+
+// Returns a naive byte count for an object when serialized
+// I was using an external library for this (object-sizeof), but getting results
+// off by a factor of ~2. This seems to be working for now. -DB
+const BLOCKLISTED_KEYS = new Set(['parent', '_call', '_return']);
```
Comment (kgilpin) [inline src/appmapBuilder.js:153]: BLOCKLISTED or BLACKLISTED?

**Review by kgilpin — CHANGES_REQUESTED** [review]
One probably typo. Also, can you describe the new functionality in the README?

File: src/appmapBuilder.js (hunk 1 above)
Comment (dustinbyrne) [reply src/appmapBuilder.js:153]: "Blocklist" was intended here

**Review by sashahonchar — APPROVED** [review]

**Comment by dustinbyrne** [comment]
There was some file duplication due to casing changes in file names. I've just fixed it.

**Pull request merged**